docs/dev/adding-new-platform.md

```diff
@@ -0,0 +1,198 @@
+2. Ensure that RHCoS has any necessary platform specific code to communicate with the platform (for instance, on Azure the instance must periodically health check) - see [cloud support tracker on Fedora CoreOS for more info](https://github.com/coreos/fedora-coreos-tracker/issues/95).
+3. Have a RHCoS image with the appropriate oemID tag set.
+
+There is a script that assists you in converting the generic VM image to have a specific oemID set in the [coreos-assembler repo as gf-oemid](https://github.com/coreos/coreos-assembler/blob/master/src/gf-oemid). See the instructions there to create an image with the appropriate ID.
+
+Once you have uploaded the image to your platform, and the machine stays up, you can begin porting the installer to have a minimal IPI.
+
+
+Continuous Integration
+----------------------
+
+To enable a new platform, require a core continuous integration testing loop that verifies that new changes do not regress our support for the platform. The minimum steps required are:
+
+1. Have an infrastructure that can receive API calls from the OpenShift CI system to provision/destroy instances
+2. Support at minimum 3 concurrent clusters on that infrastructure as "per release image" testing (https://origin-release.svc.ci.openshift.org) that verify a release supports that platform
+3. Also support a per-PR target that can be selectively run on the installer, core, and operator repositories in OpenShift in order to allow developers to test incremental changes to those components
+
+No PR will be merged to openshift/installer for platform support that cannot satisfy the above steps.
+
+
+Naming
+------
+
+The platform name will be part of our public API and must go through standard API review. The name
+should be consistent with common usage of the platform and be recognizable to a consumer.
+
+The following names for platforms are good examples of what is expected:
+
+* Amazon Web Services -> `aws` or `AWS`
+* Google Cloud Platform -> `gcp` or `GCP`
+* Azure -> `azure` or `Azure`
+* Libvirt -> `libvirt` or `Libvirt`
+* OpenStack -> `openstack` or `OpenStack`
+
+
+Enable Provisioning
+-------------------
+
+Since CI testing requires the ability to provision via an API, we define the basic path for supporting a platform as having a minimal provisioning path in the OpenShift installer. Not all platforms we support will have full infrastructure provisioning supported, but the basic path must be invokable via Go code in openshift-install before a platform can be certified. This ensures we have at least one path to installation.
+
+The OpenShift installer has normal and hidden provisioners. The hidden provisioners are explicitly unsupported for production use but are supported for testing. 
+
+1. Add a new hidden provisioner
+2. Define the minimal platform parameters that the provisioner must support
+3. Use Terraform or direct Go code to provision that platform via the credentials provided to the installer.
```

Comment:
"provision" -> "provision and destroy"?  One benefit of Terraform is that it makes centralized bootstrap teardown fairly straightforward, although you could certainly switch on the platform to invoke platform-specific Go bootstrap-teardown code.  And we need to destroy resources for `destroy cluster` to keep the account from filling with cruft, although that doesn't need to be as specific as bootstrap teardown.